aws/data_instance_profiles.go

```diff
@@ -2,36 +2,81 @@ package aws
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"regexp"
 	"strings"
 
 	"github.com/databricks/databricks-sdk-go"
 	"github.com/databricks/terraform-provider-databricks/common"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"golang.org/x/exp/slices"
 )
 
+func instanceProfileMatchesFilter(ip *instanceProfileData, filter *instanceProfileFilter) bool {
+	var ipMap map[string]interface{}
+	m, _ := json.Marshal(ip)
+	_ = json.Unmarshal(m, &ipMap)
+	val := ipMap[filter.Name]
+	stringVal := fmt.Sprint(val)
+	re := regexp.MustCompile(filter.Pattern)
```

Comment:
It's better to use `.Compile` and check for the error. Also, it's very inefficient - it's better to check validity of the regexp earlier, around line 46, compile regexp, and pass it explicitly. Otherwise we'll compile it on each function invocation.